### PR TITLE
Adjustments for pyscope on real hardware

### PR DESCRIPTION
**Pull request by pgriffin17**
Solves #125 by implementing a working exposure time logging property in the ascom_camera.py script in my exptime branch and updating the observatory class to use this, which results in the last user-input exposure time being logged to the FITS header if the ASCOM LastExposureDuration function is not implemented in the ASCOM driver.

This builds off of #124 .

### CODE REVIEW

**Comment by pgriffin17** [comment]
- Adds new ASCOM camera properties of `_last_exposure_duration` and `_last_exposure_time` 
    - These hold temporary exposure duration and image start time values in the case that the camera does not support actual readout of these parameters (as is the case for ZWO ASI290MM Mini).
- Adds LastInputExposureDuration property to ASCOM camera driver
    - This exposes `_last_exposure_duration` as set by the user.
- Currently the Observatory class has checks on which value to place for EXPTIME and EXPOSURE (this should probably be moved to the ASCOM Camera driver
    - Also adds CAMTIME to FITS header, which is True if the EXPTIME/EXPOSURE is set from the camera's own time or if it is using the user's input time. 
    
- Changes _last_exposure_start_time in maxim driver to use dt.utcnow() instead of time.time().

**Comment by pgriffin17** [comment]
I've now added a function to ASCOM Camera that will set the image data type appropriately depending on the camera settings. Documentation is shown below:
image

**Comment by pgriffin17** [comment]
Solves #125, #128, and allows pyscope to function on real minimal hardware setups.

- Tested on Losmandy Gemini (1) G11 German Equatorial mount along with ZWO ASI290MM Mini camera.

ASCOM Cameras now have their image arrays transposed by default to match their standard orientation, as well as having their data type set to 16 bit by default, or based off of their MaxADU (as in the documentation or in the screenshot above). This keeps the Observatory class the same as far as the `ImageArray` property is concerned.

Changelog:

#### ASCOM Camera related features
- Adds new ASCOM camera properties of `_last_exposure_duration` and `_last_exposure_time`
    - These hold temporary exposure duration and image start time values in the case that the camera does not support actual readout of these parameters (as is the case for ZWO ASI290MM Mini).
- Adds LastInputExposureDuration property to ASCOM camera driver
    - This exposes `_last_exposure_duration` as set by the user.
- Adds new ASCOM camera properties of `_image_data_type`, and `_DoTranspose`
    - These store the optimal data type for the images (as set by `SetImageDataType()`) and whether the `ImageArray` should be transposed.
- Currently the Observatory class has checks on which value to place for EXPTIME and EXPOSURE (this should probably be moved to the ASCOM Camera driver)
    - Also adds CAMTIME to FITS header, which is True if the EXPTIME/EXPOSURE is set from the camera's own time or if it is using the user's input time. 
    
#### `Observatory` class optimizations
- The `Observatory` class previously had several calls to self.camera.ImageArray within the `save_last_image()` function. These have now been reduced to one call that is stored in a variable and referenced several times.

#### Camera driver optimizations
- Changes _last_exposure_start_time in maxim driver and ASCOM camera driver to use dt.utcnow() instead of time.time().

#### Get Real Hardware Working by changing `_read_out_kwargs`
- To get real hardware working, adjusted location setting lines to ensure that devices are connected before setting mount parameters
- Also cleared up some Cover Calibrator if statements
- Part of the rotator section caused errors, so added a check to ensure the rotator driver was assigned before checking some kwargs.


Next up is to move EXPTIME/EXPOSURE checks (on using the camera's own exposure time, if it exists, or the user input exposure time) from the `Observatory` class to the ASCOM Camera driver.

**Comment by pgriffin17** [comment]
Now I have moved EXPTIME/EXPOSURE checks (on using the camera's own exposure time, if it exists, or the user input exposure time) from the `Observatory` class to the ASCOM Camera driver.

**Comment by pgriffin17** [comment]
This pull request:

closes #125 
closes #128

File: pyscope/observatory/ascom_camera.py
Comment (WWGolay) [inline pyscope/observatory/ascom_camera.py:1]: Could we switch from datetime to astropy Time objects maybe?

File: pyscope/observatory/maxim.py
Comment (WWGolay) [inline pyscope/observatory/maxim.py:1]: Same here, astropy Time instead maybe?

**Review by WWGolay — CHANGES_REQUESTED** [review]

File: pyscope/observatory/ascom_camera.py
Comment (pgriffin17) [reply pyscope/observatory/ascom_camera.py:1]: Fixed in https://github.com/WWGolay/pyscope/pull/127/commits/614a7bdd62eddacac5bcbd0e8c3d759937606ef4

**Review by pgriffin17 — COMMENTED** [review]
Updated to use astropy.Time.now() instead of datetime

**Review by WWGolay — APPROVED** [review]

**Pull request merged**